### PR TITLE
[StructuralMechanicsApplication] Using `AtomicAdd` in `BaseSolidElement`

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

Using `AtomicAdd` in `BaseSolidElement`. Also adding some missing documentation and some minor clean up.

**🆕 Changelog**

- [Using `AtomicAdd` in `BaseSolidElement`](https://github.com/KratosMultiphysics/Kratos/commit/901f626886826b507294a261faedc9b130588470)
- [Some minor and documentation changes](https://github.com/KratosMultiphysics/Kratos/commit/e134215233ade50aa2b0301ec1df5d82e5f9e00d)

### CODE REVIEW

**Comment by loumalouomega** [comment]
Ups @AlejandroCornejo I wanted to modief the structural elements and I ended modiying yours :P

File: applications/FemToDemApplication/custom_elements/base_solid_element.cpp [hunk 1]
```diff
@@ -1305,14 +1305,14 @@ int  BaseSolidElement::Check( const ProcessInfo& rCurrentProcessInfo ) const
...
-
-        KRATOS_CHECK_DOF_IN_NODE(DISPLACEMENT_X, rnode)
-        KRATOS_CHECK_DOF_IN_NODE(DISPLACEMENT_Y, rnode)
-        KRATOS_CHECK_DOF_IN_NODE(DISPLACEMENT_Z, rnode)
+        const Node& r_node = this->GetGeometry()[i];
+        KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(DISPLACEMENT,r_node)
+//         KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(VELOCITY,r_node)
+//         KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(ACCELERATION,r_node)
+
+        KRATOS_CHECK_DOF_IN_NODE(DISPLACEMENT_X, r_node)
```
Comment (AlejandroCornejo) [inline applications/FemToDemApplication/custom_elements/base_solid_element.cpp:1313]: is it better like this?

File: applications/FemToDemApplication/custom_elements/base_solid_element.h [hunk 2]
```diff
@@ -875,8 +876,30 @@ class KRATOS_API(FEM_TO_DEM_APPLICATION) BaseSolidElement
...
-    double GetRayleighBeta(const Properties& rProperties, const ProcessInfo& rCurrentProcessInfo);
+    /**
+     * @brief Get the Rayleigh alpha parameter.
+     * @details This function calculates and returns the Rayleigh alpha parameter based on the provided properties and current process information.
+     * @param rProperties The properties needed for the calculation.
+     * @param rCurrentProcessInfo The current process information.
+     * @return The Rayleigh alpha parameter.
+     */
+    double GetRayleighAlpha(
+        const Properties& rProperties,
```
Comment (AlejandroCornejo) [inline applications/FemToDemApplication/custom_elements/base_solid_element.h:887]: did you intend to change the elements from the `FemToDemApp`?

**Review by AlejandroCornejo — CHANGES_REQUESTED** [review]

File: applications/FemToDemApplication/custom_elements/base_solid_element.h (hunk 2 above)
Comment (loumalouomega) [reply applications/FemToDemApplication/custom_elements/base_solid_element.h:887]: By error :(

File: applications/FemToDemApplication/custom_elements/base_solid_element.cpp (hunk 1 above)
Comment (loumalouomega) [reply applications/FemToDemApplication/custom_elements/base_solid_element.cpp:1313]: It is the notation

**Pull request closed without merging**